dataprofiler/profilers/column_profile_compilers.py

```diff
@@ -40,41 +41,90 @@ def _create_profile(self, df_series, options=None):
                     
-                    warning_msg += "\n\nFor labeler errors, try installing "
-                    warning_msg += "the extra ml requirements via:\n\n"
-                    warning_msg += "$ pip install dataprofiler[ml] --user\n\n"
-
-                    warnings.warn(warning_msg, RuntimeWarning, stacklevel=2)
+                self._profiles[col_profile_type.col_type] = \
+                    col_profile_type(df_series.name, options=col_options)
+                
+                # MULTIPROCESSING
+                single_process_flag = True
+                if multiprocess_flag:
+                    single_process_flag = False
+                    try:                        
+                        f = pool.apply_async(
+                            self._profiles[col_profile_type.col_type].update,
+                            (df_series,)
+                        )
+                        multi_process_dict[col_profile_type.col_type] = f
+                    except Exception as e:
+                        # Occurs when object cannot be pickled
+                        single_process_flag = True
+
+                # Add to single process list to execute later
+                if single_process_flag:
+                    single_process_list.append(col_profile_type.col_type)
+
+        for col in multi_process_dict.keys():
+            try:
+                f = multi_process_dict[col]
+                returned_profile = f.get()
+                if returned_profile is not None:
+                    self._profiles[col] = returned_profile                
+            except Exception as e:
+                # Attempt again as a single process
+                single_process_list.append(col_profile_type.col_type)
+                    
+        # Close pool for new tasks        
+        pool.close()
+        
+        # Wait for all workers to complete
+        pool.join()
+
+        for col_profile_type_col_type in single_process_list:
+            try:
+                self._profiles[col_profile_type_col_type].update(df_series)
+            except Exception as e:
+                import warnings
+                warning_msg = "\n\n!!! WARNING Partial Profiler Failure !!!\n\n"
+                warning_msg += "Profiling Type: {}".format(col_profile_type_col_type)
+                warning_msg += "\nException: {}".format(type(e).__name__)
+                warning_msg += "\nMessage: {}".format(e)
+
+                # This is considered a major error
+                if type(e).__name__ == "ValueError":
+                    raise ValueError(e)
+            
+                warning_msg += "\n\nFor labeler errors, try installing "
+                warning_msg += "the extra ml requirements via:\n\n"
+                warning_msg += "$ pip install dataprofiler[ml] --user\n\n"
+                
+                warnings.warn(warning_msg, RuntimeWarning, stacklevel=2)
```

Comment:
can we add those warning messages to the multiprocess tasks?

Reply:
I'm unclear on what you mean.

Reply:
I think he means that currently the warnings only appear for tasks that are single process, and he wants to add them for when you iterate through the multiprocess ones, but @AnhTruong if you notice, when an exception occurs in multiprocess, the task gets added to single process, so in those cases if something truly were wrong it would go from multi_process_dict -> single_process_list -> warning, so I think its covered.

Reply:
I think the intent is that if multiprocess fails, it switches over to single anyways and if that fails will get this error so @lettergram  didn't feel the need to duplicate it if it was just a fault of multiprocessing failing, which I think is fair.

Reply:
I believe @ChrisWallace2020 is right in his assessment: `multi_process_dict -> single_process_list -> warning`

Reply:
Ok, makes sense